src/pcre2_compile_class.c

```diff
@@ -1828,7 +1773,7 @@ code += 32 / sizeof(PCRE2_UCHAR);
 
 DONE:
 *pcode = code;
-return TRUE;
+return pptr - 1;
```

Comment:
Is there a tiny bit of code smell to returning `ptr - 1` here? The last META code we saw might be a two-item one (such as POSIX + code, or BIGVALUE + codepoint). So, the -1 doesn't point to anything meaningful.

It appears you've done it, because the existing callers take the return pointer and always add a fixed +1 to it, and it's simpler in some ways to leave the callers as-is.

Just a suggestion. If the code works out neater this way, that's fine. Maybe the function should have a note in its doc at the top that you must not take the return value and treat it the start of a META code.

Reply:
It reverts the ++ here.

https://github.com/PCRE2Project/pcre2/blob/master/src/pcre2_compile_class.c#L1141

We read the token which terminates the loop. We don't process the token, just read its type.